### ISSUE TITLE
Can't change the name in the IDP Mapper configuration after it has been created

### ISSUE THREAD

**Issue by ahus1** [keycloak/keycloak#34967]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

I created a mapper. After I had saved it, I was unable to change the name. 
This is strange as in the DB it is saved with an ID.

This might be a regression in 131aeedb98fb8723410a68bd246accf0135ac128, as before there wasn't such a condition. 

The fix would be simple as it would only remove this line, still I'd like the core-clients team to verify this first: 

https://github.com/keycloak/keycloak/blob/d2e8092c7f72c58967042329360588cb3c263c6a/js/apps/admin-ui/src/identity-providers/add/AddMapperForm.tsx#L47

### Version

main

### Regression

- [X] The issue is a regression

### Expected behavior

I want to change the name after I have saved the mapper

### Actual behavior

I can't change the name after I saved it initially

### How to Reproduce?

Create an IDP (for example of the Keycloak type), and then create any mapper

**Comment by martin-kanis** [comment]
True, we have now unique IDs for IDP mappers. Previously id == name and both were not editable. We just need to pay attention to caching because the mapper name is part of the cached key. If we update name, the current invalidation will invalidate `idp-alias+new-name` which cannot be cached yet.
